apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts

```diff
@@ -72,14 +85,23 @@ function updateEditor() {
 			// Disable href navigation
 			e.preventDefault();
 
-			const shouldOpenNewFlow = ! isMobile || ( isMobile && isNewLaunchMobile );
+			// Clicking on the persisten "Launch" button (when added to the UI)
+			// would normally open the control launch flow by redirecting the
+			// page to `launchUrl`.
+			// But if the site was created via Gutenboarding (/new),
+			// and potentially depending on the browser's viewport, the control
+			// launch flow replaced by a new "Complete setup" flow, appering in a
+			// modal on top of the edittor (no redirect needed)
+			const shouldOpenNewFlowModal =
+				isGutenboarding && ( ! isMobileViewport || ( isMobileViewport && isNewLaunchMobile ) );
 
 			recordTracksEvent( 'calypso_newsite_editor_launch_click', {
-				is_new_flow: shouldOpenNewFlow,
+				// TODO: 'is_new_flow' flag — is it still relevant?
+				is_new_flow: shouldOpenNewFlowModal,
```

Comment:
@razvanpapadopol What changes should we make in terms of analytics for the Persistent launch button?